python/ray/data/read_api.py

```diff
@@ -559,7 +568,7 @@ def read_text(
     parallelism: int = 200,
     arrow_open_stream_args: Optional[Dict[str, Any]] = None,
     meta_provider: BaseFileMetadataProvider = DefaultFileMetadataProvider(),
-    partition_filter: PathPartitionFilter = None,
+    partition_filter: PathPartitionFilter = FileExtensionFilter("txt"),
```

Comment:
I don't think this is a good idea for text files, how about accepting everything by default here?

Reply:
Hmm ok yeah good point!